### DISCUSSION TITLE
Generate supergraphs

### DISCUSSION THREAD

**Discussion by hemals24** [graphql-java/graphql-java#2688] [Q&A]
From command line, I can execute the rover cli supergraph command to generate a supergraph from sub graphs. Can I do something similar using the Java lib. I am looking at ways to create a supergraph using Java libraries.

**Comment by bbakerman** [comment]
graphql-java itself is a base graphql engine and is not involved in Apollo Federation, which is a higher level style of composing a system.

So there is no code in graphql-java  related to Federation.

https://github.com/apollographql/federation-jvm is the Apollo produced JVM lib but I am not sure if it has the super graph generation you are after.  Apollo have bet strongly on Rust (rover) for this composition.

Maybe others know more than me.

**Comment by guykatz** [comment]
can anyone share their experience using graphql-java with federation-jvm?
in a microservice environment, there is no running away from federating/stiching things into a single public API
at one point we will all probably need to federate somehow...
so if anyone can share an experience it would be great!
